api/src/v2/Registration/Uploader/LayDeputyshipUploader.php

```diff
@@ -111,21 +109,32 @@ private function updateReportTypes(): LayDeputyshipUploader
         $reports = $this->reportRepository->findAllActiveReportsByCaseNumbersAndRole($caseNumbers, User::ROLE_LAY_DEPUTY);
 
         try {
+            /** @var Report $currentActiveReport */
             foreach ($reports as $currentActiveReport) {
                 $reportCaseNumber = strtolower($currentActiveReport->getClient()->getCaseNumber());
                 $currentActiveReportId = $currentActiveReport->getId();
                 /** @var PreRegistration $preRegistration */
                 $preRegistration = $this->preRegistrationEntriesByCaseNumber[$reportCaseNumber];
                 $determinedReportType = PreRegistration::getReportTypeByOrderType($preRegistration->getTypeOfReport(), $preRegistration->getOrderType(), PreRegistration::REALM_LAY);
 
-                if ($currentActiveReport->getType() != $determinedReportType) {
-                    $currentActiveReport->setType($determinedReportType);
-                    $this->reportsUpdated[] = $reportCaseNumber;
+                // For Dual Cases, deputy uid needs to match for the report type to be updated
+                if (PreRegistration::DUAL_TYPE == $preRegistration->getHybrid()) {
+                    if ($currentActiveReport->getClient()->getUsers()[0]->getDeputyNo() == $preRegistration->getDeputyUid()) {
+                        if ($currentActiveReport->getType() != $determinedReportType) {
+                            $currentActiveReport->setType($determinedReportType);
+                            $this->reportsUpdated[] = $reportCaseNumber;
+                        }
+                    }
+                } else {
+                    if ($currentActiveReport->getType() != $determinedReportType) {
+                        $currentActiveReport->setType($determinedReportType);
+                        $this->reportsUpdated[] = $reportCaseNumber;
+                    }
```

Comment:
Not as bad here as the previous comment but I still think doing the below would keep the if's statements to just a 2 nested statement.

```
// For Dual Cases, deputy uid needs to match for the report type to be updated
if (PreRegistration::DUAL_TYPE != $preRegistration->getHybrid()) {
    if ($currentActiveReport->getType() != $determinedReportType) {
        $currentActiveReport->setType($determinedReportType);
        $this->reportsUpdated[] = $reportCaseNumber;
        continue;
    }
}

if ($currentActiveReport->getClient()->getUsers()[0]->getDeputyNo() == $preRegistration->getDeputyUid()) {
    if ($currentActiveReport->getType() != $determinedReportType) {
        $currentActiveReport->setType($determinedReportType);
        $this->reportsUpdated[] = $reportCaseNumber;
    }
}
```